### DISCUSSION TITLE
VideoCapture.API.DShow does not work after the update Windows

### DISCUSSION THREAD

**Discussion by mr-kot** [emgucv/emgucv#922] [General]
After updating windows 10 and 11, the program that works with the camera stopped running. 
The problem turned out to be in a line of code:
`_capture = new VideoCapture(i, VideoCapture.API.DShow);`
As I understand it, Microsoft has dropped support for DirectShow in Windows updates. Everything works on computers without the latest updates.
If you use MSMF, the program works, but it takes a long time to start:
`_capture = new VideoCapture(i, VideoCapture.API.Msmf);`
I tried disabling acceleration. It has almost no effect on the launch speed.
`_capture = new VideoCapture(i, VideoCapture.API.Msmf, new Tuple<CapProp, int>(CapProp.HwAcceleration, (int)VideoAccelerationType.None));`
Even when using MSMF, the frame resolution is always the maximum. This code does not change the resolution of the frame:
```
_capture.Set(CapProp.FrameWidth, width);
_capture.Set(CapProp.FrameHeight, height);
```
How do I continue using DsShow after Windows update? Or use MSMF to be able to change the resolution?

**Comment by amacsmith** [comment]
I have the EXACT same issue have you found any work arounds @mr-kot?

**Comment by mr-kot** [comment]
Unfortunately, I have not found a solution. We have to use msmf with maximum resolution. This mode loads the processor heavily. Probably because of the high resolution. I also noticed that with video cameras of other models, the resolution may be the minimum from the list of supported cameras. And there's no way to change it either.
I also found a problem with msmf. The index of the video camera received by the command:
`DsDevice[] SystemCamereas = DsDevice.GetDevicesOfCat(FilterCategory.VideoInputDevice);`
It may not match the camera index in the command:
`_capture = new VideoCapture(index, VideoCapture.API.Msmf);`
Especially if there are more than two video cameras in the system.

**Comment by KurtKomell** [comment]
I have the same problem. It throws a cvextern.dll error.
